### ISSUE TITLE
[3.18.0 bug/regression] (old) usernames with a space lead to "Server not availabe" ("Server not verfügbar") while reading files

### ISSUE THREAD

**Issue by Wikinaut** [nextcloud/android#9331]
### Steps to reproduce
Nextcloud Android app 3.18.0 installed - and reinstalled
Device-specific password authentication (linked newly)
re-added useraccount on the Android

Useraccount is of form "givenname familyname" (old useraccount with a space in it) example: "Test Test" 

### Actual behaviour
Files are not listed anymore and/or are not refreshed, message is "Server nicht verfügbar"
but: SofortUpload and manuall upload still  work!

It looks to me, that the space in the username is stopping certain read actions in the app, but not all. For example, the accunt image is shown.

Logfile shows ```Login failed: 'Test Test'``` everytime when the Android app is started.

### Environment data
Android version: 10

Device model: HUAWEI P Smart 2017

Nextcloud app version:
3.18.0 and beta

Nextcloud server version:
22.2.3
and also on 22.2.0

**Comment by Fischi5** [comment]
Hi
 I have the same problem since the last Thursday. I administration a NextCloud instance for our club. All users with an Android smartphone have the same problem, "Server Not Availabe".

On the iPhone this is no problem, as well as in the web browser or on Windows clients. Everything works up to Android.

**Comment by maxl99** [comment]
Same Problem with Nextcloud Server Version 20 up to 22.2.3

Android Apk 3.17.0 works fine.
Since 3.18.0 no files get listed and the message "Server not available" will be shown for usernames with spaces.

**Comment by w-e-w** [comment]
I maybe experiencing a similar issue
I also have a **space in my username**
I'm seeing the "**server not available**" message in folders

it seems that it's still have some connection to the server
I can still receive notifications or see activities

I'm **able to "read" as in download files,**
but I am **not able to refresh the table of contents file / folders**

meaning if I have open the folder before so the app has cache the list of files, I will be able to download the file
but for a newly added file or folders I have not opened before, I will not be able to see the file, and unable to access them

**Comment by AlvaroBrey** [comment]
Likely an unintended consequence of https://github.com/nextcloud/android-library/pull/729, although we'll have to do some verification to see if it's the case.

**Comment by lapor-kris** [comment]
We are also facing the same problem. Usernames with space cannot connect to the server.

**Comment by Wikinaut** [comment]
When will the new Nextcloud Android App version be deployed via Google Play/App Store?

**Comment by Sgt-Nukem** [comment]
After wondering why the heck my password database on my phone doesn't have the latest entries from the desktop anymore, I assumed the Keepass app was at fault.

I then realized Seedvault didn't back up my phone for almost two weeks!!

Seeing "Server not available" in the NextCloud app while logging in via browser (on the same phone) worked as expected brought me to #9366 - and then here...

**TL;DR: This is the kind of issue that is supposed to be fixed with a hotfix asap and not delayed until the next proper release.**

**Comment by EasyNetDev** [comment]
Hi,

I got the same issue as @Wikinaut for more than 1 month.
I thought is a bug in Nextcloud Server and this is the bug report I filled out: #29888.
Seems this bug is resolved with version 20211202 of dev client version.

**Comment by JujuBickoille** [comment]
This buf is back again on 3.20.0

**Comment by eleminer** [comment]
jap, same problem
